pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/DestinationLookup.java

```diff
@@ -223,8 +223,9 @@ public void lookupDestinationAsync(@PathParam("property") String property, @Path
                                 log.debug("[{}] Lookup result {}", fqdn.toString(), lookupResult);
                             }
 
-                            LookupData lookupData = lookupResult.getLookupData();
-                            if (lookupResult.isRedirect()) {
+                            checkArgument(lookupResult.isPresent());
```

Comment:
Though, actually, it should fall back in the `exceptionally()` path and correctly propagate the exception on the future